### ISSUE TITLE
Error: http://deis.d.domain.com does not appear to be a valid Deis controller.

### ISSUE THREAD

**Issue by olalonde** [deis/deis#4892]
After a fresh stateless install of deis v1.12.2 on AWS, all my units are "active/running" but when I try to register to the controller, I'm getting:

```
$ deis register http://deis.d.domain.com
Error: http://deis.d.domain.com does not appear to be a valid Deis controller.
Make sure that the Controller URI is correct and the server is running.
```

I made sure that I had a CNAME that points to my ELB on both `d.domain.com` and `*.d.domain.com`.

The controller does not seem to receive any requests:

```bash
$ fleetctl journal -f deis-controller
-- Logs begin at Mon 2016-02-01 01:08:05 UTC. --
Feb 01 03:22:18 ip-10-21-1-171.ec2.internal sh[17352]: [2016-02-01 03:22:18 +0000] [126] [INFO] Listening at: http://0.0.0.0:8000 (126)
Feb 01 03:22:18 ip-10-21-1-171.ec2.internal sh[17352]: [2016-02-01 03:22:18 +0000] [126] [INFO] Using worker: sync
Feb 01 03:22:18 ip-10-21-1-171.ec2.internal sh[17352]: [2016-02-01 03:22:18 +0000] [130] [INFO] Booting worker with pid: 130
Feb 01 03:22:18 ip-10-21-1-171.ec2.internal sh[17352]: [2016-02-01 03:22:18 +0000] [131] [INFO] Booting worker with pid: 131
Feb 01 03:22:18 ip-10-21-1-171.ec2.internal sh[17352]: [2016-02-01 03:22:18 +0000] [132] [INFO] Booting worker with pid: 132
Feb 01 03:22:18 ip-10-21-1-171.ec2.internal sh[17352]: [2016-02-01 03:22:18 +0000] [133] [INFO] Booting worker with pid: 133
Feb 01 03:22:18 ip-10-21-1-171.ec2.internal sh[17352]: [2016-02-01 03:22:18 +0000] [134] [INFO] Booting worker with pid: 134
Feb 01 03:22:19 ip-10-21-1-171.ec2.internal sh[17352]: Publishing DB state to etcd...
Feb 01 03:22:19 ip-10-21-1-171.ec2.internal sh[17352]: Done Publishing DB state to etcd.
Feb 01 03:22:19 ip-10-21-1-171.ec2.internal sh[17352]: deis-controller running...

```

and the router is getting requests which I don't really understand (AWS PROXY protocol):

```bash
$ fleetctl journal -f deis-router@1
-- Logs begin at Sun 2016-01-31 23:25:49 UTC. --
Feb 01 03:21:56 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - 2016/02/01 03:21:56 [notice] 44#0: signal process started
Feb 01 03:21:56 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - deis-router running...
Feb 01 03:22:21 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - 2016/02/01 03:22:21 [notice] 64#0: signal process started
Feb 01 03:22:32 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - 2016/02/01 03:22:32 [notice] 72#0: signal process started
Feb 01 03:24:51 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - [01/Feb/2016:03:24:51 +0000] - 10.21.1.164 - - - 400 - "PROXY TCP4 98.248.246.95 10.0.10.222 62505 80" - 323 - "-" - "-" - "" - - - - - - - 0.089
Feb 01 03:24:56 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - [01/Feb/2016:03:24:56 +0000] - 10.21.1.164 - - - 400 - "PROXY TCP4 98.248.246.95 10.0.10.222 62507 80" - 323 - "-" - "-" - "" - - - - - - - 0.091
Feb 01 03:24:59 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - [01/Feb/2016:03:24:59 +0000] - 10.21.1.164 - - - 400 - "PROXY TCP4 98.248.246.95 10.0.10.222 62509 80" - 323 - "-" - "-" - "" - - - - - - - 0.102
Feb 01 03:25:17 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - [01/Feb/2016:03:25:17 +0000] - 10.21.1.164 - - - 400 - "PROXY TCP4 98.248.246.95 10.0.10.222 62536 80" - 323 - "-" - "-" - "" - - - - - - - 0.086
Feb 01 03:30:35 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - [01/Feb/2016:03:30:35 +0000] - 10.21.1.164 - - - 400 - "PROXY TCP4 98.248.246.95 10.0.10.222 62662 80" - 323 - "-" - "-" - "" - - - - - - - 0.091
Feb 01 03:30:38 ip-10-21-1-186.ec2.internal sh[7332]: [INFO] - [01/Feb/2016:03:30:38 +0000] - 10.21.1.164 - - - 400 - "PROXY TCP4 98.248.246.95 10.0.10.222 62671 80" - 323 - "-" - "-" - "" - - - - - - - 0.086

```

I'm not sure what that `10.21.1.164` nor `10.0.10.222` represents.

```
$ fleetctl list-machines
MACHINE         IP              METADATA
4b07b764...     10.21.1.186     controlPlane=true,dataPlane=true,routerMesh=true
7b8ab575...     10.21.1.171     controlPlane=true,dataPlane=false,routerMesh=true
cb55969e...     10.21.2.173     controlPlane=true,dataPlane=true,routerMesh=true
ea3764fe...     10.21.2.172     controlPlane=true,dataPlane=true,routerMesh=true
```

```
$ etcdctl ls /deis/router/hosts
/deis/router/hosts/10.21.2.172
/deis/router/hosts/10.21.1.171
/deis/router/hosts/10.21.1.186
```

```
$ fleetctl list-units
UNIT                    MACHINE                 ACTIVE  SUB
deis-builder.service    cb55969e.../10.21.2.173 active  running
deis-controller.service 7b8ab575.../10.21.1.171 active  running
deis-logger.service     7b8ab575.../10.21.1.171 active  running
deis-logspout.service   4b07b764.../10.21.1.186 active  running
deis-logspout.service   7b8ab575.../10.21.1.171 active  running
deis-logspout.service   cb55969e.../10.21.2.173 active  running
deis-logspout.service   ea3764fe.../10.21.2.172 active  running
deis-publisher.service  4b07b764.../10.21.1.186 active  running
deis-publisher.service  cb55969e.../10.21.2.173 active  running
deis-publisher.service  ea3764fe.../10.21.2.172 active  running
deis-registry@1.service 4b07b764.../10.21.1.186 active  running
deis-router@1.service   4b07b764.../10.21.1.186 active  running
deis-router@2.service   7b8ab575.../10.21.1.171 active  running
deis-router@3.service   ea3764fe.../10.21.2.172 active  running
```

Help :(

**Comment by olalonde** [comment]
Oh, looks like the node where controller started is not part of the etcd cluster. That makes sense because I added it later.

```
core@ip-10-21-1-171 ~ $ etcdctl cluster-health
member 2f73055b916d695b is healthy: got healthy result from http://10.21.1.186:2379
member 38920251877c13a9 is healthy: got healthy result from http://10.21.2.173:2379
member 480f6772af64d3e0 is healthy: got healthy result from http://10.21.2.172:2379
cluster is healthy
core@ip-10-21-1-171 ~ $ fleetctl list-units | grep controller
deis-controller.service 7b8ab575.../10.21.1.171 active  running
core@ip-10-21-1-171 ~ $
```

**Comment by olalonde** [comment]
Manually changing my AWS ELB to use HTTP instead of TCP seems to have fixed the problem.

Before:
<img width="817" alt="screenshot 2016-01-31 20 27 28" src="https://cloud.githubusercontent.com/assets/416585/12708848/90d24ddc-c859-11e5-99ef-b8cde8cfbee7.png">

After:
<img width="794" alt="screenshot 2016-01-31 20 29 41" src="https://cloud.githubusercontent.com/assets/416585/12708847/90ce5f7e-c859-11e5-9ddb-a2cabd16a676.png">

**Comment by olalonde** [comment]
Oh ok, I think I figured out what happened! 

At some point I uninstalled deis and did a `etcdctl rm --recursive /deis` to make sure I had a clean state. I re-followed all the [stateless install steps](http://docs.deis.io/en/latest/managing_deis/running-deis-without-ceph/) and installed/started the stateless platform. 

What I didn't notice was that the `provision-aws-cluster.sh` [wrote that key](https://github.com/deis/deis/commit/4f634ed1275c47c60ec5070124c681923525b78a#diff-f037290b5e19752a3db2b7a5e63e6b43R139) to etcd: `deisctl config router set proxyProtocol=1` and I had lost it when I wiped `/deis`.

tl;dr: fixed problem with `deisctl config router set proxyProtocol=1`
